rfd/0136-modern-signature-algorithms.md

```diff
@@ -0,0 +1,406 @@
+---
+authors: Nic Klaassen (nic@goteleport.com)
+state: draft
+---
+
+# RFD 136 - Modern Signature Algorithms
+
+## Required Approvers
+
+TODO
+
+## What
+
+Teleport should support modern key types and signature algorithms, currently
+only RSA2048 keys are supported with the PKCS#1 v1.5 signature scheme.
+This applies to CA keys and client (user/host) keys, but each can/will be
+addressed individually.
+
+## Why
+
+Modern algorithms like ECDSA and Ed25519 offer better security properties with
+smaller keys that are faster to generate and sign with.
+Some of the more restrictive security policies are starting to reject RSA2048
+(e.g. [RHEL 8's FUTURE policy](https://access.redhat.com/articles/3642912)).
+
+## Details
+
+### Algorithms
+
+These algorithms are being considered for support:
+
+#### RSA
+
+Private key sizes: 2048, 3072, 4096
+
+Signature algorithms: PKCS#1 v1.5, (maybe PSS for TLS and JWT? SSH does not support it)
+
+Digest/hash algorithms: SHA512 for SSH, SHA256 for TLS
```

Comment:
We currently support SHA256/512 by default. The only reason why we still allow the use of SHA-1 is OpenSSH 7.4 is installed by default in CentOS 7, which fails on anything newer than SHA-1. 
Adding support for Ed25519 would potentially solve this problem, as OpenSSH claims support for it starting 6.8, but I've personally never tested it.

Reply:
I have been using `Ed25519` with SSH for a couple years now, no issues

Reply:
For a long time I still used RSA since AWS only recently let customers user ed25519 keys. Luckily they now fully support ed25519 https://aws.amazon.com/about-aws/whats-new/2021/08/amazon-ec2-customers-ed25519-keys-authentication/ https://aws.amazon.com/about-aws/whats-new/2022/01/ed25519-keys-authentication-ec2-instance-connect/

Reply:
I also try to use the ed25519 key everywhere, but I still have RSA key in my `~/.ssh` directory as not all places support non-RSA keys. AFAIR GitLab still doesn't have support for them (maybe they added it recently).

Reply:
looks like GitLab supports Ed25519 now https://docs.gitlab.com/ee/user/ssh.html#ed25519-ssh-keys